### PR TITLE
Add digest manifest

### PR DESCRIPTION
**Pull request by dgnorton**

### CODE REVIEW

**Comment by hercules-influx** [comment]
During a run of [megacheck](https://github.com/dominikh/go-tools) the following issues were discovered:
```/tmp/364419185/src/github.com/influxdata/influxdb/cmd/store/query/query.go:390:5: this value of line is never used (SA4006)
/tmp/364419185/src/github.com/influxdata/influxdb/cmd/store/query/query.go:404:5: this value of line is never used (SA4006)
```

File: tsdb/engine/tsm1/digest.go [hunk 1]
```diff
@@ -99,9 +107,141 @@ func DigestWithOptions(dir string, opts DigestOptions, w io.WriteCloser) error {
...
+		return false, fmt.Sprintf("Digest belongs to another shard. Manually copied?: manifest_dir=%s, shard_dir=%s", mfest.Dir, dir)
+	}
+
+	// See if the number of tsm files matches what's listed in the manifest.
+	if len(files) != len(mfest.Entries) {
+		return false, fmt.Sprintf("Number of tsm files differ: engine=%d, manifest=%d", len(files), len(mfest.Entries))
+	}
+
+	// See if all the tsm files match the manifest.
+	sort.StringSlice(files).Sort()
```
Comment (e-dard) [inline tsdb/engine/tsm1/digest.go]: `sort.Strings`?

File: tsdb/engine/tsm1/engine.go [hunk 2]
```diff
@@ -286,48 +286,44 @@ func (e *Engine) Digest() (io.ReadCloser, int64, error) {
...
 			if err != nil {
-				log.Info("Digest aborted, can't stat existing digest", zap.Error(err))
-				f.Close()
+				log.Info("Digest aborted, couldn't stat digest file", zap.Uint64("shard", e.id), zap.Error(err))
 				return nil, 0, err
 			}
 
-			log.Info("Digest cache fresh", zap.Time("engine_last_modified", engineLastMod),
-				zap.Time("digest_cache_last_modified", fi.ModTime()), zap.Int64("size", fi.Size()))
+			log.Info("Digest is fresh", zap.Uint64("shard", e.id), zap.String("path", digestPath))
```
Comment (e-dard) [inline tsdb/engine/tsm1/engine.go]: User `logger.Shard` for the shard id.

File: tsdb/engine/tsm1/engine.go [hunk 3]
```diff
@@ -286,48 +286,44 @@ func (e *Engine) Digest() (io.ReadCloser, int64, error) {
...
-		if err := f.Close(); err != nil {
-			log.Info("Digest aborted, problem closing digest", zap.Error(err))
-			return nil, 0, err
-		}
-
-		log.Info("Digest cache stale", zap.Time("engine_last_modified", engineLastMod),
-			zap.Time("digest_cache_last_modified", fi.ModTime()), zap.Int64("size", fi.Size()))
 	}
 
+	log.Info("Digest stale", zap.Uint64("shard", e.id), zap.String("reason", reason))
```
Comment (e-dard) [inline tsdb/engine/tsm1/engine.go]: `logger.Shard`

**Pull request merged**